internal/sdkprovider/service/vpc/organization_vpc_test.go

```diff
@@ -0,0 +1,177 @@
+			},
+			{
+				// import state
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				// test the data source
+				Config: newComposition().
+					Add(organizationVPCResource, map[string]interface{}{
+						"resource_name": "test_org_vpc",
+						"cloud_name":    "aws-eu-west-1",
+						"network_cidr":  "10.0.0.0/24",
+					}).
+					Add(organizationVPCDataSource, map[string]interface{}{
+						"resource_name": "vpc_ds",
+					}).MustRender(t),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(dataSourceName, "organization_id", resourceName, "organization_id"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "cloud_name", resourceName, "cloud_name"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "network_cidr", resourceName, "network_cidr"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "state", resourceName, "state"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "create_time", resourceName, "create_time"),
+					resource.TestCheckResourceAttr(dataSourceName, "state", string(organizationvpc.VpcStateTypeActive)),
+					resource.TestCheckResourceAttrSet(dataSourceName, "organization_vpc_id"),
+					resource.TestCheckResourceAttrPair(dataSourceName, "organization_vpc_id", resourceName, "organization_vpc_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckOrganizationVPCResourceDestroy(s *terraform.State) error {
+	ctx := context.Background()
+
+	c, err := acc.GetTestGenAivenClient()
+	if err != nil {
+		return fmt.Errorf("error initializing Aiven client: %w", err)
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != organizationVPCResource {
+			continue
+		}
+
+		orgID, vpcID, err := schemautil.SplitResourceID2(rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("error splitting resource ID: %w", err)
+		}
+
+		orgVPC, err := c.OrganizationVpcGet(ctx, orgID, vpcID)
+		if common.IsCritical(err) {
+			return fmt.Errorf("error fetching VPC (%q): %w", vpcID, err)
+		}
+
+		if orgVPC != nil {
+			return fmt.Errorf("VPC (%q) for organization (%q) still exists", vpcID, orgID)
+		}
+	}
+
+	return nil
+}
+
+func preSetOrgVPCTemplates(t *testing.T) *acc.TemplateRegistry {
+	t.Helper()
+
+	var registry = acc.NewTemplateRegistry(organizationVPCResource)
+
+	registry.MustAddTemplate(t, "organization_data", `
+data "aiven_organization" "foo" {
+	name = "{{ .organization_name }}"
+}`)
+
+	registry.MustAddTemplate(t, organizationVPCResource, `
+resource "aiven_organization_vpc" "{{ .resource_name }}" {
+    organization_id = data.aiven_organization.foo.id
+    cloud_name     = "{{ .cloud_name }}"
+    network_cidr   = "{{ .network_cidr }}"
+}`)
+
+	registry.MustAddTemplate(t, organizationVPCDataSource, `
+data "aiven_organization_vpc" "{{ .resource_name }}" {
+    organization_id = data.aiven_organization.foo.id
+    organization_vpc_id          = split("/", aiven_organization_vpc.test_org_vpc.id)[1]
```

Comment:
Please refer to `aiven_organization_vpc.test_org_vpc.organization_vpc_id`

Reply:
Fixed